toolchain/check/eval.cpp

```diff
@@ -17,6 +17,104 @@
 
 namespace Carbon::Check {
 
+namespace {
+// Information about an eval block of a specific that we are currently building.
+struct SpecificEvalInfo {
+  // The region within the specific whose eval block we are building.
+  SemIR::GenericInstIndex::Region region;
+  // The work-in-progress contents of the eval block.
+  llvm::ArrayRef<SemIR::InstId> values;
+};
+
+// Information about the context within which we are performing evaluation.
+struct EvalContext {
```

Comment:
Noting how much of this is functions, should it be a class? Examining member uses, it looks like `context` and `specific_id` are part of the API, but `file` and `specific_eval_info` are not.

Though, thinking about this from a different perspective, had you considered making the static functions in eval.cpp be members of this class? I think we use the `Context` object elsewhere because we have a lot of different files we want to compose access from, but here, we have a scoped use-case. It'd also avoid the need to pass `EvalContext` as a parameter.

Without a `class`, `file` and `sem_ir()` in particular look like an odd match to me: why an accessor to return a public member?

Reply:
My intent is to make this into a class, but I'm trying to keep this PR minimal. Will send a follow-up with that change.